docs/_docs/maintaining/releasing-a-new-version.md

```diff
@@ -0,0 +1,81 @@
+---
+title: "Releasing a new version"
+---
+
+**This guide is for maintainers.** These special people have **write access** to one or more of Jekyll's repositories and help merge the contributions of others. You may find what is written here interesting, but it’s definitely not for everyone.
+{: .note .info }
+
+The most important thing to understand before making a release is that there's no need to feel nervous. Most things are revertable, and even if you do publish an incomplete gem version, we can always skip that one. Don't hestitate to contact the other maintainers if you feel unsure or don't know what to do next.
+
+### Bump the version
+
+The only important place you need to manually bump the version is in `lib/jekyll/version.rb`. Adjust that, and everything else should work fine.
+
+### Update the history document
+
+Replace the first header of the history document with a version milestone. This looks like the following:
+
+```diff
+-## HEAD
++## 3.7.1 / 2018-01-25
+```
+
+Adjust the version number and the date. The `## HEAD` heading will be regenerated next time a pull request is merged.
+
+Once you've done this, update the website by running the following command:
+
+```sh
+bundle exec rake site:generate
+```
+
+This updates the website's changelog, and pushes the versions in various other places.
+
+It's recommended that you go over the `History.markdown` file manually one more time, in case there are any spelling errors or such. Feel free to fix those manually, and after you're done generating the website changelog, commit your changes.
+
+## Write a release post
+
+In case this isn't done already, you can generate a new release post using the included `rake` command:
+
+```sh
+bundle exec rake site:releases:new[3.8.0]
+```
+
+where `3.8.0` should be replaced with the new version. Then, write the post. Be sure to thank all of the collaborators and maintainers who have contributed since the last release. You can generate a log of their names using the following command:
+
+```sh
+git shortlog -sn master...v3.7.2
+```
+
+where, again `v3.7.2` is the last release. Be sure to open a pull request for your release post.
+
+### Push the version
+
+Before you do this step, make sure the following things are done:
+
+- You have permission to push a new gem version to RubyGems
+- You're logged into RubyGems on your command line
+- A release post has been prepared, and is ideally already live
```

Comment:
We should make sure the release post is merged in so that the bundled site in `jekyll-docs` sees the release post.